File: lottery/loto.rb
def new_draw
    balls = (1..49).to_a
    draw_balls=[]
    6.times do
        draw_ball = balls.sample
        draw_balls << draw_ball
        balls = balls - [draw_ball]
    end
    draw_balls.sort!
end

14000000.times do |draw_number|
    draw=new_draw
    puts "Draw Number: " + draw_number.to_s + ", Draw: " + draw.to_s if draw == [1, 2, 3, 4, 5, 6]
    puts "Draw Number: " + draw_number.to_s + ", Draw: " + draw.to_s if draw == [1, 12, 31, 42, 45, 46]
end

File: lottery/loto.go
package main

import "fmt"
import "math/rand"

func new_draw() [6]int {
    var balls [49]int;
    for index := 0; index < 49; index++ {
       balls[index]=index+1
    }
    var draw_balls [6]int
    for draw_number := 0; draw_number < 6; draw_number++ {
      draw_balls[draw_number] = rand.Intn(49)
    }
    return draw_balls
}

func main() {
    var expected = [6]int{1, 2, 3, 4, 5, 6}
    var draw [6]int
    
    for i := 0; i < 14000000; i++ {
        draw = new_draw()
        //fmt.Println(draw)
        if (draw == expected) {
    	    fmt.Println("[1, 2, 3, 4, 5, 6]")
        }
    }
}